### DISCUSSION TITLE
Can I using composables in the server/api, why they not auto import?

### DISCUSSION THREAD

**Discussion by xby020** [nuxt/nuxt#16164] [Questions] [answered]
LIke This:
```
// composables/useTest.ts
export const useFoo = () => {
  return 'foo';
}

// server/api/foo.ts
export default defineEventHandler((event) =>
  const foo = useFoo();
  return {
    foo:foo
  }
})
```
then the `useFoo` will be not defined, but if I manually import `useFoo` to foo.ts it it behaves normally.

**Comment by Diizzayy** [comment, marked as answer]
@xby020 Composables are only meant to be used in the Vue portion of your app, hence why they're not auto imported to nitro (the server) and why manually importing works, however though doing so is not advised.

**Reply by oripka** [reply]
So what is the recommended way for server apis to import reusable code?
